pkg/nuclide/file-tree/lib/FileTreeController.js

```diff
@@ -740,6 +748,43 @@ class FileTreeController {
     this._openDialog(<FileDialogComponent {...props} />);
   }
 
+  openDuplicateDialog(): void {
+    var selection = this._getSelectedEntryAndDirectoryAndRoot();
+    if (!selection) {
+      return;
+    }
+
+    if(selection.entry.isDirectory()) {
+      return;
+    }
+    var message = (
+      <div>
+        <div>Enter the new name for the file in the root:</div>
+        <div>{path.normalize(selection.root.getPath() + '/')}</div>
+      </div>
+    );
+
+    var {entry, root} = selection;
+
+    var FileDialogComponent = require('./FileDialogComponent');
```

Comment:
Can you move this into `_openDialog`? It's a bit goofy that it was copied in the first place, we can do a little cleanup here. `_openDialog` can take `props` as an argument instead of a `ReactComponent`.